cmd/list_metadata.go

```diff
@@ -0,0 +1,121 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	e "github.com/cloudposse/atmos/internal/exec"
+	"github.com/cloudposse/atmos/pkg/config"
+	list "github.com/cloudposse/atmos/pkg/list"
+	l "github.com/cloudposse/atmos/pkg/logger"
+	"github.com/cloudposse/atmos/pkg/schema"
+)
+
+// listMetadataCmd lists metadata across stacks
+var listMetadataCmd = &cobra.Command{
+	Use:   "metadata",
+	Short: "List metadata across stacks",
+	Long:  "List metadata information across all stacks",
+	Example: "atmos list metadata\n" +
+		"atmos list metadata --query .component\n" +
+		"atmos list metadata --format json\n" +
+		"atmos list metadata --stack '*-dev-*'\n" +
+		"atmos list metadata --stack 'prod-*'",
+	Run: func(cmd *cobra.Command, args []string) {
+		// Check Atmos configuration
+		checkAtmosConfig()
+
+		// Initialize logger from CLI config
+		configAndStacksInfo := schema.ConfigAndStacksInfo{}
+		atmosConfig, err := config.InitCliConfig(configAndStacksInfo, true)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error initializing CLI config: %v\n", err)
+			return
+		}
+
+		logger, err := l.NewLoggerFromCliConfig(atmosConfig)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error initializing logger: %v\n", err)
+			return
+		}
+
+		flags := cmd.Flags()
+
+		queryFlag, err := flags.GetString("query")
+		if err != nil {
+			logger.Error(fmt.Errorf("failed to get query flag: %v", err))
```

Comment:
```suggestion
			logger.Error("failed to get query flag", "error", err)
```
Use semantic logging, not wrapping.

Update all log statements in this PR